src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -339,16 +340,20 @@ def _get_child(parent, collection_name, item_name, collection_key):
         return result
 
 
-def _get_operations_tmpl(cmd):
-    operations_tmpl = cmd.command_kwargs.get('operations_tmpl',
-                                             cmd.command_kwargs.get('command_type').settings['operations_tmpl'])
+def _get_operations_tmpl(cmd, custom_command=False):
+    if custom_command:
+        operations_tmpl = cmd.command_kwargs.get('operations_tmpl') or \
+            cmd.command_kwargs.get('custom_command_type').settings['operations_tmpl']
+    else:
+        operations_tmpl = cmd.command_kwargs.get('operations_tmpl') or \
+            cmd.command_kwargs.get('command_type').settings['operations_tmpl']
     if not operations_tmpl:
         raise CLIError("command authoring error: cmd '{}' does not have an operations_tmpl.".format(cmd.name))
     return operations_tmpl
 
 
-def _get_client_factory(_, kwargs):
-    command_type = kwargs.get('command_type', None)
+def _get_client_factory(_, custom_command=False, **kwargs):
+    command_type = kwargs.get('custom_command_type', None) if custom_command else kwargs.get('command_type', None)
```

Comment:
Why isn't custom_command just another kwarg?

Reply:
`kwargs` are passed along. This ensures that `custom_command` is only used where it is intended; in the `generic_*` related methods..